client/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java

```diff
@@ -170,76 +175,101 @@ private WebMarkupContainer updateRealmContent(final RealmTO realmTO) {
         if (realmTO == null) {
             return content;
         }
-        content.addOrReplace(new Realm("body", realmTO, getPageReference()) {
+        content.addOrReplace(new RealmConcrete("body", realmTO, getPageReference()));
+        return content;
+    }
 
-            private static final long serialVersionUID = 8221398624379357183L;
+    private WebMarkupContainer updateRealmContent(final RealmTO realmTO, final int selectedIndex) {
+        if (realmTO == null) {
+            return content;
+        }
+        content.addOrReplace(new RealmConcrete("body", realmTO, getPageReference(), selectedIndex));
+        return content;
+    }
 
-            @Override
-            protected void onClickTemplate(final AjaxRequestTarget target) {
-                templates.setTargetObject(realmTO);
-                templates.toggle(target, true);
-            }
+    class RealmConcrete extends Realm {
```

Comment:
I don't understand why you have created `RealmConcrete` instead of putting your additions directly into `Realm`